mpas_analysis/shared/regions/compute_region_masks_subtask.py

```diff
@@ -0,0 +1,262 @@
+            self.featureList = get_feature_list(self.config,
+                                                self.geojsonFileName)
+
+        mpasMeshName = self.config.get('input', 'mpasMeshName')
+
+        # first, see if we have cached a mask file name in the region masks
+        # directory
+        regionMaskDirectory = build_config_full_path(self.config,
+                                                     'regions',
+                                                     'regionMaskDirectory')
+        self.maskFileName = '{}/{}_{}.nc'.format(regionMaskDirectory,
+                                                 mpasMeshName,
+                                                 self.outFileSuffix)
+
+        if os.path.exists(self.maskFileName):
+            self.maskExists = True
+        else:
+            # no cached mask file, so let's see if there's already one in the
+            # masks subfolder of the output directory
+
+            maskSubdirectory = build_config_full_path(self.config, 'output',
+                                                      'maskSubdirectory')
+            self.maskFileName = '{}/{}_{}.nc'.format(maskSubdirectory,
+                                                     mpasMeshName,
+                                                     self.outFileSuffix)
+
+            self.maskExists = os.path.exists(self.maskFileName)
+
+        # }}}
+
+    def run_task(self):  # {{{
+        '''
+        Compute the requested climatologies
+        '''
+        # Authors
+        # -------
+        # Xylar Asay-Davis
+
+        if self.maskExists:
+            return
+
+        self.logger.info('Creating masks file {}'.format(self.maskFileName))
+
+        with xr.open_dataset(self.restartFileName) as dsRestart:
+            dsRestart = mpas_xarray.subset_variables(dsRestart,
+                                                     ['lonCell', 'latCell'])
+            latCell = numpy.rad2deg(dsRestart.latCell.values)
+
+            # transform longitudes to [-180, 180)
+            lonCell = numpy.mod(numpy.rad2deg(dsRestart.lonCell.values) + 180.,
+                                360.) - 180.
+
+        # create shapely geometry for lonCell and latCell
+        cellPoints = [shapely.geometry.Point(x, y) for x, y in
+                      zip(lonCell, latCell)]
+
+        nCells = len(cellPoints)
+
+        masks = []
+        regionNames = []
+        nChar = 0
+        self.logger.info('  Computing masks from {}...'.format(
+                self.geojsonFileName))
+        with open(self.geojsonFileName) as f:
+            featureData = json.load(f)
+
+        for feature in featureData['features']:
+            name = feature['properties']['name']
+            if name not in self.featureList:
+                continue
+
+            self.logger.info('      {}'.format(name))
+
+            shape = shapely.geometry.shape(feature['geometry'])
+            mask = numpy.array([shape.contains(point) for point
```

Comment:
This may potentially be slow...

Reply:
Here's a way to plot shapely MultiPolygons as paths in matplotlib.  That might be a way to efficiently rasterize a region at high resolution so that it could then be used to determine whether MPAS cell centers are in or out.  I'll look into this.  If it's not too complicated and costly, I'll do this instead.

https://sgillies.net/2010/04/06/painting-punctured-polygons-with-matplotlib.html

Reply:
Hmm, on a bit more investigation, `descartes` actually does this stuff behind the scenes, which is why we use it for plotting in `geometric_features`.  Anyway, something to look into before this PR gets merged.